crates/sui-graphql-client/src/lib.rs

```diff
@@ -121,6 +121,30 @@ impl<T> Page<T> {
     }
 }
 
+/// Pagination direction.
+pub enum Direction {
+    Forward,
+    Backward,
+}
+
+/// Pagination options for querying the GraphQL server. It defaults to forward pagination with the
+/// GraphQL server's default items per page limit.
+pub struct PaginationFilter<'a> {
+    direction: Direction,
+    cursor: Option<&'a str>,
+    limit: Option<i32>,
+}
+
+impl Default for PaginationFilter<'_> {
+    fn default() -> Self {
+        Self {
+            direction: Direction::Forward,
+            cursor: None,
+            limit: None,
+        }
+    }
+}
+
```

Comment:
This should be a simpler way to achieve the same thing:

```suggestion
/// Pagination direction.
#[derive(Default)]
pub enum Direction {
    #[default]
    Forward,
    Backward,
}

/// Pagination options for querying the GraphQL server. It defaults to forward pagination with the
/// GraphQL server's default items per page limit.
#[derive(Default)
pub struct PaginationFilter<'a> {
    direction: Direction,
    cursor: Option<&'a str>,
    limit: Option<i32>,
}
```

Reply:
Thanks!